src/affine_decomposition.jl

```diff
@@ -60,16 +48,59 @@ Perform the compression of an [`AffineDecomposition`](@ref)
 corresponding to ``o = B^\\dagger O B``.
 """
 function compress(ad::AffineDecomposition, basis::RBasis)
-    AffineDecomposition(compress.(ad.terms, Ref(basis)), ad.coefficient_map)
+    matrixel = compress.(ad.terms, Ref(basis.snapshots))  # ⟨ψᵢ|O|ψⱼ⟩
+    rbterms = map(m -> basis.vectors' * m * basis.vectors, matrixel)  # transform to RB
+    (; rb=AffineDecomposition(rbterms, ad.coefficient_map),
+     raw=AffineDecomposition(matrixel, ad.coefficient_map))
 end
+
+"""
+    compress(ad::AffineDecomposition{<:Matrix,<:Function},
+             basis::RBasis; symmetric_terms=false)
+
+Perform compression for an [`AffineDecomposition`](@ref) with terms
+with two indices (double-sum observables), including an option to
+exploit the possible symmetry of terms ``O_{r,r'} = O_{r',r}``,
+such that only the necessary compressions are computed.
+"""
+function compress(ad::AffineDecomposition{<:Matrix,<:Function},
+                  basis::RBasis; symmetric_terms=false)
+    # TODO: replace symmetric_terms by some generalized Symmetric type in the long run
+    if symmetric_terms
+        ⪒(a, b) = (size(ad.terms, 1) > size(ad.terms, 2)) ? ≥(a, b) : ≤(a, b)
```

Comment:
Great idea, but ⪒ is quite tough to read. Pick a better symbol (or just a function name)

Reply:
how about `is_nonredundant`

Reply:
Yes that would be a good name -- earlier I just struggled to find a fitting name, so I thought I would just take the fitting symbol. I'll change that to `is_nonredundant` then.